haystack/document_stores/types/filter_policy.py

```diff
@@ -36,10 +36,41 @@ def from_str(filter_policy: str) -> "FilterPolicy":
         return policy
 
 
+def is_legacy(filter_item: Dict[str, Any]) -> bool:
+    """
+    Check if the given filter is a legacy filter.
+
+    :param filter_item: The filter to check.
+    :returns: True if the filter is a legacy filter, False otherwise.
+    """
+    return "operator" not in filter_item
+
+
+def is_comparison_filter(filter_item: Dict[str, Any]) -> bool:
+    """
+    Check if the given filter is a comparison filter.
+
+    :param filter_item: The filter to check.
+    :returns: True if the filter is a comparison filter, False otherwise.
+    """
+    return all(key in filter_item for key in ["field", "operator", "value"])
+
+
+def is_logical_filter(filter_item: Dict[str, Any]) -> bool:
+    """
+    Check if the given filter is a logical filter.
+
+    :param filter_item: The filter to check.
+    :returns: True if the filter is a logical filter, False otherwise.
+    """
+    return "operator" in filter_item and "conditions" in filter_item
+
+
 def apply_filter_policy(
     filter_policy: FilterPolicy,
     init_filters: Optional[Dict[str, Any]] = None,
     runtime_filters: Optional[Dict[str, Any]] = None,
+    default_logical_operator: str = "AND",
```

Comment:
Missing doc string. Also do we know what are possible values this can be already, if so could we make this an Enum or a Literal type?